### DISCUSSION TITLE
Manual triggers with workflow_dispatch on tag

### DISCUSSION THREAD

**Discussion by el7cosmos** [community/community#26642] [Actions]
<p>At the moment, manually running workflow from UI can select which branch to run. Can we also get to select a tag?</p>

**Comment by brushknight** [comment]
<p>This would be super nice. Waiting for this feature :joystick:</p>

**Comment by gsanchezgavier** [comment]
<p>+1 on this feature implementation</p>

**Comment by Simran-B** [comment]
<p>This is actually supported in the API, but not in the UI.</p>
<aside class="quote quote-modified" data-post="4" data-topic="130561">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/workflow-dispatch-from-a-tag-in-actions-tab/130561/4">Workflow Dispatch from a tag in actions tab</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    Also see <a href="https://github.community/t/workflow-dispatch-cant-set-ref-to-commit-sha/138132" class="inline-onebox">Workflow dispatch - can't set ref to commit SHA</a> 
A commit hash isn’t supported, but tags are accepted by the REST API. 
curl \
  -d '{"ref":"{tag}"}' \
  -H 'Content-Type: application/json' \
  -H 'Accept: application/vnd.github.v3+json' \
  -H 'Authorization: token {access-token}' \
  'https://api.github.com/repos/{owner}/{repo}/actions/workflows/{workflow_id}/dispatches'

The expected reply in case of a successful dispatch is HTTP/1.1 204 No Content without payload. 
Depending on the …
  </blockquote>
</aside>

**Comment by peterotoole** [comment]
<p>+1 on this feature too</p>

**Comment by braynmquarterone** [comment]
<p>+1 It would be nice to have!</p>

**Comment by roelblyweert** [comment]
<p>+1 on this feature too!</p>

**Comment by cschleiden** [comment, marked as answer]
<p>You can now select tags from the UI.</p>

**Reply by ravindrabajpai** [reply]
I tried it. Upon selecting the tag and click 'Run Workflow' but nothing happens.  This tag is on the master(default) branch.
![image](https://github.com/community/community/assets/1445512/58e81067-1e43-405a-8ac9-9c6708ce101f)

Please let me know if there is anything wrong here.

**Reply by ravindrabajpai** [reply]
Found the issue and answer here - https://stackoverflow.com/questions/75404816/github-actions-workflow-workflow-dispatch-running-a-workflow-manually-most-of.

**Comment by PawlikMateusz** [comment]
<p>Hello, how can run this from the UI for the old tags? I’m getting the message:</p>
<pre><code class="lang-auto">Workflow does not exist or does not have a workflow_dispatch trigger in this tag.[Learn more about manual workflows](https://docs.github.com/actions/)
</code></pre>

**Comment by Simran-B** [comment]
<p>What is an “old” tag?</p>

**Comment by PawlikMateusz** [comment]
<p>I meant every existing tag</p>

**Comment by Simran-B** [comment]
<p>Does the state of the code at the selected tag have a workflow with a <code>workflow_dispatch</code> trigger? If not, then I’m afraid that you can’t manually start a workflow run unless you add one and update the tag.</p>

**Comment by ghost** [comment]
Any updates on this ? It would be great to manually run a workflow from any tag in a deployment use case.

**Reply by cschleiden** [reply]
The workflow needs to exist in the given ref.

**Reply by bholzer** [reply]
This is a pretty unfortunate shortfall. I want to consider tags immutable, but that's not realistic if GitHub actions forces me to modify tags to deploy old code with new processes.
